public/lib/content-service.js

```diff
@@ -1,24 +1,39 @@
 import angular from 'angular';
 
 import './composer-service';
+import './media-atom-maker-service'
 import './http-session-service';
 import './user';
 import './visibility-service';
+import './feature-switches';
 
-angular.module('wfContentService', ['wfHttpSessionService', 'wfVisibilityService', 'wfDateService', 'wfFiltersService', 'wfUser', 'wfComposerService'])
-    .factory('wfContentService', ['$rootScope', '$log', 'wfHttpSessionService', 'wfDateParser', 'wfFormatDateTimeFilter', 'wfFiltersService', 'wfComposerService',
-        function ($rootScope, $log, wfHttpSessionService, wfDateParser, wfFormatDateTimeFilter, wfFiltersService, wfComposerService) {
+angular.module('wfContentService', ['wfHttpSessionService', 'wfVisibilityService', 'wfDateService', 'wfFiltersService', 'wfUser', 'wfComposerService', 'wfMediaAtomMakerService'])
+    .factory('wfContentService', ['$rootScope', '$log', 'wfHttpSessionService', 'wfDateParser', 'wfFormatDateTimeFilter', 'wfFiltersService', 'wfComposerService', 'wfMediaAtomMakerService', 'config', 'wfFeatureSwitches',
+        function ($rootScope, $log, wfHttpSessionService, wfDateParser, wfFormatDateTimeFilter, wfFiltersService, wfComposerService, wfMediaAtomMakerService, config, wfFeatureSwitches) {
 
-            var httpRequest = wfHttpSessionService.request;
+            const httpRequest = wfHttpSessionService.request;
 
             class ContentService {
                 getTypes() {
+
+                    return wfFeatureSwitches.getSwitch("support-atoms").then( (isActive) => {
+                        var basicTypes = {
+                            "article": "Article",
+                            "liveblog": "Live blog",
+                            "gallery": "Gallery",
+                            "interactive": "Interactive",
+                            "picture": "Picture"
+                        };
+
+                        if (isActive) basicTypes['atom'] = 'Atom';
+
+                        return basicTypes;
```

Comment:
could use `Object.assign` to avoid the mutation:

```js
const basicTypes = {'article': 'Article', ...}
return isActive
  ? Object.assign({}, basicTypes, {'atom': 'Atom'})
  :  basicTypes;
```